### PR TITLE
[RN] Consolidate eventTypes registry with view configs

### PR DESCRIPTION
**Pull request by sebmarkbage**
We already have one stateful module that contains all the view config.
We might as well store the event types there too. That way the shared
state is compartmentalized (and I can move it out in a follow up PR).

The view config registry also already has an appropriate place to call
processEventTypes so now we no longer have to do it in RN.

Will follow up with a PR to RN to remove that call.

### CODE REVIEW

File: packages/react-native-renderer/src/ReactNativeViewConfigRegistry.js [hunk 1]
```diff
@@ -49,6 +90,7 @@ export function get(name: string): ReactNativeBaseComponentViewConfig {
     );
     viewConfigCallbacks.set(name, null);
     viewConfig = callback();
+    processEventTypes(viewConfig);
```
Comment (sebmarkbage) [inline packages/react-native-renderer/src/ReactNativeViewConfigRegistry.js]: This is what lets us avoid exposing this module.

**Comment by numkhing** [comment]
I do not know what to do with it, but I do not know how to do it. I do not
> have any malicious intentions, I'm sorry, I accept all the faults in the
> actions I have done to tight.
>

File: packages/react-native-renderer/src/ReactFabric.js [hunk 2]
```diff
@@ -84,7 +83,6 @@ const ReactFabric: ReactNativeType = {
     // Used as a mixin in many createClass-based components
     NativeMethodsMixin,
     // Used by react-native-github/Libraries/ components
-    ReactNativeBridgeEventPlugin, // requireNativeComponent
```
Comment (bvaughn) [inline packages/react-native-renderer/src/ReactFabric.js]: We can also delete the forwarding `scripts/rollup/shims/react-native/ReactNativeBridgeEventPlugin.js` now?

Comment (bvaughn) [reply packages/react-native-renderer/src/ReactFabric.js]: Should we just get rid of the `ReactNativeBridgeEventPlugin` entirely now and just move `extractEvents` over into the view config registry as well?

**Comment by bvaughn** [comment]
Gotcha.

File: packages/react-native-renderer/src/ReactFabric.js (hunk 2 above)
Comment (sebmarkbage) [reply packages/react-native-renderer/src/ReactFabric.js]: Good catch on the shim. I'll delete it.

**Comment by janicduplessis** [comment]
@sebmarkbage @bvaughn We're using `ReactNativeBridgeEventPlugin.processEventTypes` in [react-native-gesture-handler](https://github.com/kmagiera/react-native-gesture-handler) to register a custom event type for views (see [here](https://github.com/kmagiera/react-native-gesture-handler/blob/master/GestureHandler.js#L44) and [here](https://github.com/kmagiera/react-native-gesture-handler/blob/master/GestureHandler.js#L289)). There isn't a particular native component associated with these events so we cannot use requireNativeComponent (we used to register a dummy native component as a hack but now events are registered lazily so it doesn't work).

Is there an alternative I'm not seeing or could we bring this back as public'ish API (as public as importing `react-native/Libraries/Renderer/shims/ReactNativeBridgeEventPlugin` gets 😄)?

**Comment by gaearon** [comment]
This is a bit ironic because `shims` was added as a temporary measure to expose private APIs to RN itself. Didn’t even occur to me third-party libs would attempt to import those 😛

**Comment by janicduplessis** [comment]
Yea it is a very low level library and RN doesn't export anything to be able to do that so some creativity was needed. A solution that could also work is to expose a root HOC that renders a native component that registers the view types but it seems like quite a bit of overhead just to register the events (both the extra API in react-native-gesture-handler and the runtime overhead of the extra view).

If we decide to bring this back we could add it as extra RN public API so it is less likely to break in the future, although I'm not familiar enough with this API to know how stable it is. Something like an undocumented `UNSTABLE_registerExtraEventType` might be good.

**Pull request merged**